### PR TITLE
OpenHarmony Device Interface Adaptation

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/platform/openharmony/modules/Screen.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     @brief Get current display stats.
+     @return bool, is displaying stats or not.
+     */
+    bool isDisplayStats() override;
+
+    /**
+     @brief set display stats information.
+     */
+    void setDisplayStats(bool isShow) override;};
```
Comment (minggo) [inline native/cocos/platform/openharmony/modules/Screen.h]: error `};` ?

File: native/cocos/platform/openharmony/modules/System.cpp [hunk 2]
```diff
@@ -38,16 +39,64 @@ std::string System::getDeviceModel() const {
...
+    } else if (languageStr == "tr") {
+        return ISystem::LanguageType::TURKISH;
+    } else if (languageStr == "uk") {
+        return ISystem::LanguageType::UKRAINIAN;
+    } else if (languageStr == "ro") {
+        return ISystem::LanguageType::ROMANIAN;
+    } else if (languageStr == "bg") {
+        return ISystem::LanguageType::BULGARIAN;
+    }
+    return ISystem::LanguageType::ENGLISH;
```
Comment (minggo) [inline native/cocos/platform/openharmony/modules/System.cpp:83]: JS returns result with `return i18n.getSystemLanguage();`. Then why need to implement in C++?

Comment (qiuguohua) [reply native/cocos/platform/openharmony/modules/System.cpp:83]: To keep the interface consistent. Both interfaces are exported for use by the cocos engine.
And the value returned by i18n.getSystemLanguage() is a little different than we expected.

Comment (minggo) [reply native/cocos/platform/openharmony/modules/System.cpp:83]: > And the value returned by i18n.getSystemLanguage() is a little different than we expected.

If so, why use it in js?

Comment (qiuguohua) [reply native/cocos/platform/openharmony/modules/System.cpp:83]: i18n.getSystemLanguage() is called for C++, which converts the expected values and returns them to the ts engine.
In addition, i18n is an internal module and interface of openharmony.

**Review by minggo — APPROVED** [review]

**Pull request merged**